_posts/2022-07-15-stork-load-balancing.adoc

```diff
@@ -0,0 +1,233 @@
+----
+<dependency>
+ <groupId>io.smallrye.stork</groupId>
+ <artifactId>stork-load-balancer-least-requests</artifactId>
+</dependency>
+----
+
+You also need to configure the load balancer of your service:
+
+[source, properties]
+----
+stork.my-service.load-balancer=least-requests
+----
+
+If you call the service sequentially, it always picks the same instance:
+
+image::/assets/images/posts/stork-load-balancing/least-requests-seq.png[align="center"]
+
+Indeed, the number of inflight requests for the service is always 0, as it waits for the response before calling it another time. 
+However, when you call the service concurrently, this strategy becomes a lot more relevant:
+
+image::/assets/images/posts/stork-load-balancing/least-requests-con.png[align="center"]
+
+As the green service is the fastest, it gets called more frequently as inflight requests decrease quickly. 
+Thus, while the blue and yellow services still have inflight calls, the green service is back to 0 and selected.
+
+This strategy is convenient when you have services with an API having different response times. 
+For example, you can have endpoints answering quickly and endpoints doing a lot more work, thus taking more time. 
+
+This strategy has one drawback. 
+If you have many service instances, you need to iterate over the whole set to find the one to pick. 
+
+== The power of two random choices
+
+This strategy extends the _least-request_ one and addresses the drawback mentioned above. 
+Instead of iterating over the whole list of instances, it picks two instances randomly and selects the one with the fewer inflight requests.
+
+To use this strategy, you need the following dependency:
+
+[source, xml]
+----
+<dependency>
+ <groupId>io.smallrye.stork</groupId>
+ <artifactId>stork-load-balancer-power-of-two-choices</artifactId>
+</dependency>
+----
+
+You also need to configure the load balancer for the service:
+
+[source, properties]
+----
+stork.my-service.load-balancer=power-of-two-choices
+----
+
+The results are similar to the _least-request_ as we have only three instances:
+
+image::/assets/images/posts/stork-load-balancing/power-of-two-choices.png[align="center"]
+
+However, this strategy is less expensive when you have many instances and when the iteration takes too much time. 
+In the worst case, it would pick the two most busy instances and select the one with fewer requests.
+
+== Least response time
+
+The least-response time keeps track of the response time and failures. 
+Thus, it selects the fastest instances. 
+That strategy handles failures by adding a penalty to failing instances.
+
+To use this strategy, you need the following dependency:
+
+[source, xml]
+----
+<dependency>
+ <groupId>io.smallrye.stork</groupId>
+ <artifactId>stork-load-balancer-least-response-time</artifactId>
+</dependency>
+----
+
+You also need to configure the load-balancer for the service:
+
+[source, properties]
+----
+stork.my-service.load-balancer=least-response-time
```

Comment:
The config properties are not up to date....
`stork.my-service.load-balancer` should be `stork.my-service.load-balancer.type`